File: tests/js/test_extended.js
/**
 * Test entry point for extended message tests (JavaScript).
 */

const { MESSAGE_COUNT, getMessage, checkMessage } = require('./include/extended_messages');
const { getMessageInfo } = require('../generated/js/extended-test.structframe');
const { run } = require('./include/test_harness');

const TEST_NAME = 'ExtendedMessages';
const PROFILES = 'bulk, network';

process.exit(run(MESSAGE_COUNT, getMessage, checkMessage, getMessageInfo, TEST_NAME, PROFILES));

File: tests/csharp/test_extended.cs
/**
 * Test entry point for extended message tests (C#).
 */

using System;
using StructFrame.ExtendedTest;
using StructFrameTests;

class TestExtended
{
    private const string TEST_NAME = "ExtendedMessages";
    private const string PROFILES = "bulk, network";

    public static int Main(string[] args)
    {
        return TestHarness.Run(
            args,
            ExtendedMessages.MESSAGE_COUNT,
            ExtendedMessages.GetMessage,
            ExtendedMessages.CheckMessage,
            (msgId) => MessageDefinitions.GetMessageInfo(msgId),
            TEST_NAME,
            PROFILES
        );
    }
}
